lib/plugins/aws/deploy/index.js

```diff
@@ -85,7 +87,16 @@ class AwsDeploy {
           if (this.options.noDeploy) {
             return BbPromise.resolve();
           }
-          return this.serverless.pluginManager.spawn('aws:deploy:deploy');
+
+          return BbPromise.bind(this)
+            .then(this.setBucketName)
+            .then(this.shouldBeDeployed)
```

Comment:
I would add a new event `checkForChanges` in the commands:lifecycleEvents just before `uploadArtifacts` and do the check for differences in `aws:deploy:deploy:checkForChanges`.
If the check is not disabled (by something like a `--force` switch) the event should bail out with a rejection, that tells the user that nothing has been deployed.

This will keep the outer event lifecycle simple and understandable. Imo the inner lifecycle should be started in any case, but should bail out if it is terminated due to the condition.